gems/pending/VolumeManager/MiqLvm.rb

```diff
@@ -408,13 +427,17 @@ def initialize(lvId = nil, lvName = nil, segmentCount = 0)
     @status = []
     @vgObj = nil                        # a reference to this LV's volume group
   end
+
+  def thin_segments
+    segments.select { |segment| ['thin-pool', 'thin'].include?(segment.type) }
```

Comment:
Since we call this multiple times, I would memoize it:
```
@thin_segments ||= segments.select { |segment| ['thin-pool', 'thin'].include?(segment.type) }
```